pymc_extras/statespace/models/structural/components/regression.py

```diff
@@ -172,64 +183,103 @@ def make_symbolic_graph(self) -> None:
             row_idx, col_idx = np.diag_indices(self.k_states)
             self.ssm["state_cov", row_idx, col_idx] = sigma_beta.ravel() ** 2
 
-    def populate_component_properties(self) -> None:
+    def _set_parameters(self) -> None:
         k_endog = self.k_endog
         k_endog_effective = 1 if self.share_states else k_endog
+        k_states = self.k_states // k_endog_effective
+
+        beta_parameter = Parameter(
+            name=f"beta_{self.name}",
+            shape=(k_endog_effective, k_states) if k_endog_effective > 1 else (k_states,),
+            dims=(
+                (f"endog_{self.name}", f"state_{self.name}")
+                if k_endog_effective > 1
+                else (f"state_{self.name}",)
+            ),
+            constraints=None,
+        )
+
+        if self.innovations:
+            sigma_parameter = Parameter(
+                name=f"sigma_beta_{self.name}",
+                shape=(k_states,),
+                dims=(f"state_{self.name}",),
+                constraints="Positive",
+            )
+
+            self.param_info = ParameterInfo(parameters=[beta_parameter, sigma_parameter])
+        else:
+            self.param_info = ParameterInfo(parameters=[beta_parameter])
 
+    def _set_data_info(self) -> None:
+        k_endog = self.k_endog
+        k_endog_effective = 1 if self.share_states else k_endog
         k_states = self.k_states // k_endog_effective
 
+        data_prop = Data(
+            name=f"data_{self.name}",
+            shape=(None, k_states),
+            dims=(TIME_DIM, f"state_{self.name}"),
+            is_exogenous=True,
+        )
+        self.data_info = DataInfo(data=[data_prop])
+
+    def _set_shocks(self) -> None:
         if self.share_states:
-            self.shock_names = [f"{state_name}_shared" for state_name in self.state_names]
+            shock_names = [f"{state_name}_shared" for state_name in self.state_names]
         else:
-            self.shock_names = self.state_names
+            shock_names = self.state_names
 
-        self.param_names = [f"beta_{self.name}"]
-        self.data_names = [f"data_{self.name}"]
-        self.param_dims = {
-            f"beta_{self.name}": (f"endog_{self.name}", f"state_{self.name}")
-            if k_endog_effective > 1
-            else (f"state_{self.name}",)
-        }
+        self.shock_info = ShockInfo(shocks=[Shock(name=name) for name in shock_names])
 
-        base_names = self.state_names
+    def _set_states(self) -> None:
+        self.base_names = self.state_names
```

Comment:
do this assignment in `__init__` if you want to store the base_names. You can do it after you call `super()`.

As a general rule, you shouldn't create a member variable outside of the initializer.